kedro_kubeflow/generators/one_pod_pipeline_generator.py

```diff
@@ -0,0 +1,78 @@
+import logging
+import os
+
+import kubernetes.client as k8s
+from kfp import dsl
+
+from ..auth import IAP_CLIENT_ID
+from ..utils import clean_name
+from .utils import create_params, maybe_add_params
+
+
+class OnePodPipelineGenerator(object):
+    log = logging.getLogger(__name__)
+
+    def __init__(self, config, project_name, context):
+        self.project_name = project_name
+        self.context = context
+        dsl.ContainerOp._DISABLE_REUSABLE_COMPONENT_WARNING = True
+        self.run_config = config.run_config
+        self.catalog = context.config_loader.get("catalog*")
+
+    def generate_pipeline(self, pipeline, image, image_pull_policy):
+        @dsl.pipeline(self.project_name, self.run_config.description)
+        @maybe_add_params(self.context.params)
+        def convert_kedro_pipeline_to_kfp() -> None:
+            dsl.get_pipeline_conf().set_ttl_seconds_after_finished(
+                self.run_config.ttl
+            )
+            self._build_kfp_op(pipeline, image, image_pull_policy)
+
+        return convert_kedro_pipeline_to_kfp
+
+    def _build_kfp_op(
+        self,
+        pipeline,
+        image,
+        image_pull_policy,
+    ) -> dsl.ContainerOp:
+        kwargs = {
+            "env": [
```

Comment:
Do you think it would be possible to create one function across both generators to provide kwargs based on some context? I'd like to extend it soon by adding image name -> this is required for spark to run executors with same image

Reply:
Something like `create_params`

Reply:
Actually, this can be done in the next PR easily.